xcodeproj/internal/files/linker_input_files.bzl

```diff
@@ -85,8 +85,15 @@ def _compute_primary_static_library(
     # target, but sometimes another rule creates the output and this rule
     # outputs it. So far the first library has always been the correct one.
     if objc_libraries:
-        for library in objc_libraries:
-            if library.is_source:
+        generated_libraries = [f for f in objc_libraries if not f.is_source]
```

Comment:
i'd consider calling this out specifically as a bazel 6 issue for easier removal once we drop that version

Reply:
I mean, `objc_libraries` in general will go away once we remove Bazel 6 support.

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/blob/a3a572c2ef4270796e75007ceaa10dc5e1d148e0/xcodeproj/internal/compilation_providers.bzl#L241-L252